packages/dds/tree/src/test/feature-libraries/treeCursorUtils.spec.ts

```diff
@@ -0,0 +1,264 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { compareFieldUpPaths, compareUpPaths, FieldUpPath, UpPath } from "../../core";
+// Allow importing from this specific file which is being tested:
+// eslint-disable-next-line import/no-internal-modules
+import { PrefixedPath, prefixFieldPath, prefixPath } from "../../feature-libraries/treeCursorUtils";
+import { brand } from "../../util";
+
+describe("treeCursorUtils", () => {
+    const root: UpPath = {
+        parent: undefined,
+        parentField: brand("x"),
+        parentIndex: 5,
+    };
+    const child: UpPath = {
+        parent: root,
+        parentField: brand("z"),
+        parentIndex: 10,
+    };
+    describe("prefixPath", () => {
+        it("not wrapped cases", () => {
+            assert.equal(prefixPath(undefined, undefined), undefined);
+            assert.equal(prefixPath({}, undefined), undefined);
+            assert.equal(prefixPath(undefined, root), root);
+            assert.equal(prefixPath({}, root), root);
+            assert.equal(
+                prefixPath({ indexOffset: 1, rootFieldOverride: brand("y") }, undefined),
+                undefined,
+            );
+        });
+
+        it("wrapped root", () => {
+            assert(
+                compareUpPaths(prefixPath({ indexOffset: 1 }, root), {
+                    parent: undefined,
+                    parentField: brand("x"),
+                    parentIndex: 6,
+                }),
+            );
+            assert(
+                compareUpPaths(prefixPath({ rootFieldOverride: brand("y") }, root), {
```

Comment:
I'm still not sure I understand the purpose of the rootFieldOverride. So if a cursor wraps another cursor using this prefix, does it just mean that the outer cursor moves to the same index but in the y field? And so, is it just a quick way to shift around fields and if so, why do we only allow an override at the root?

Reply:
I have updated the docs on PathRootPrefix. Does it address this well enough now?

Reply:
Yes, that helps a lot. Thank you!